src/cdk/overlay/overlay-ref.ts

```diff
@@ -45,6 +45,10 @@ export class OverlayRef implements PortalHost {
   attach(portal: Portal<any>): any {
     let attachResult = this._portalHost.attach(portal);
 
+    if (this._state.positionStrategy) {
+      this._state.positionStrategy.attach(this);
```

Comment:
Doesn't this end up being a circular dependency? The overlay ref has a reference to the strategy already and now the strategy can have a reference to the ref.